docs/current/api/975146-concepts.md

```diff
@@ -0,0 +1,126 @@
+---
+slug: /api/concepts
+---
+
+# Key Concepts
+
+## Introduction
+
+One of Dagger's most powerful features is cross-language instrumentation: teams can use each other's tools without learning each other's language. Dagger achieves this by exposing its functionality through GraphQL, which serves as a low-level language-agnostic API framework for all Dagger operations.
+
+This approach has two main benefits:
+
+- It enables CI/CD pipeline authors to use whichever language feels natural and intuitive to them, without having to think about interoperability.
+- Rather than reimplementing the same automation in every possible programming language, it only needs to be implemented once and is then reusable by all. As a result, two cooperating teams no longer necessarily have to come to consensus on a common language.
+
+This tutorial explains the key concepts you need to use the Dagger GraphQL API.
+
+:::note
+This tutorial explains some basic GraphQL concepts in the context of the Dagger GraphQL API, but it is not intended as a full-featured GraphQL tutorial. To learn more about GraphQL and how it works, read the [GraphQL tutorial](https://graphql.org/learn/).
+:::
+
+## Objects, Fields and Build Operations
+
+Let's start with a basic query:
+
+```graphql
+query {
+  container {
+    from(address: "alpine:latest") {
+      exec(args: ["apk", "info"]) {
+        stdout {
+          contents
+        }
+      }
+    }
+  }
+}
+```
+
+This query represents a very simple Dagger pipeline. In plain English, it instructs Dagger to "download the latest `alpine` container image, run the command `apk info` in that image, and print the results of the command to the standard output device".
+
+Here's an example of the command output: a list of the packages installed in the image:
+
+```graphql
+{
+ "data": {
+  "container": {
+   "from": {
+    "exec": {
+     "stdout": {
+      "contents": "alpine-baselayout-data\nmusl\nbusybox\nalpine-baselayout\nalpine-keys\nca-certificates-bundle\nlibcrypto1.1\nlibssl1.1\nssl_client\nzlib\napk-tools\nscanelf\nmusl-utils\nlibc-utils\n"
+     }
+    }
+   }
+  }
+ }
+}
+```
+
+A GraphQL schema works by defining one or more object types, and then fields (and field arguments) on those object types. The fields of an object type can themselves be objects, allowing for different entities to be logically connected with each other. API users can then perform GraphQL queries to return all or some fields of an object.
+
```

Comment:
> * It's almost comparable to a bash pipeline where you pass data along and each step manipulates it in some way

by bash pipeline you mean piping multiple commands? I guess that analogy is correct, since basically it's a pipe of transformations to the initial state. Isn't "reactive programming" (https://en.wikipedia.org/wiki/Reactive_programming) the closest term?

Reply:
I've incorporated some of @sipsma 's comments in the draft now. Thanks.